WordPress/openverse-frontend
# Use props instead of store for searchTerm (query.q)

## Fixes


Fixes #1010 by @obulat

## Description


This PR removes store access from the image and audio search result pages, and turns them again into `presentational` components, i.e., depending only on the props they get, and not the store values.
It's nice to add this change before converting the stores to Pinia so that the number of changes to review in that PR is at least a little bit lower.

## Testing Instructions

Run the app and search for images and audio. You should see the correct page titles:
**All content:**
<img width="530" alt="Screen Shot 2022-02-27 at 9 32 01 AM" src="https://user-images.githubusercontent.com/15233243/155871228-086a8520-afc8-466f-9df7-9e3640fea028.png">
**Images:**
<img width="563" alt="Screen Shot 2022-02-27 at 9 27 22 AM" src="https://user-images.githubusercontent.com/15233243/155871230-737d907c-8553-4ecd-91de-d6258b4c040d.png">

BTW, why is it 'Cat | Openverse' for  'All content' and 'Cat - Openverse' for 'Images' and 'Audio'?

There is also an unrelated change to speed up the e2e test runs: when checking that an element does not exist with `await expect(element).toHaveCount(0)`, Playwright does not detect an element, so it waits for a default time period (probably 5 seconds), which makes the test take longer. I added a timout to `toHaveCount(0)` calls to speed the tests up.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
